src/fiatExchanges/SelectProvider.tsx

```diff
@@ -0,0 +1,304 @@
+  const account = useSelector(currentAccountSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+  const [noPaymentMethods, setNoPaymentMethods] = useState(false)
+  const { flow } = route.params
+
+  const digitalAsset = {
+    [Currency.Celo]: CiCoCurrency.CELO,
+    [Currency.Dollar]: CiCoCurrency.CUSD,
+    [Currency.Euro]: CiCoCurrency.CEUR,
+  }[route.params.selectedCrypto]
+
+  const asyncProviders = useAsync(async () => {
+    if (!account) {
+      Logger.error(TAG, 'No account set')
+      return
+    }
+    try {
+      const providers = await fetchProviders({
+        userLocation,
+        walletAddress: account,
+        fiatCurrency: localCurrency,
+        digitalAsset,
+        fiatAmount: route.params.amount.fiat,
+        digitalAssetAmount: route.params.amount.crypto,
+        txType: flow === CICOFlow.CashIn ? 'buy' : 'sell',
+      })
+
+      const rawLegacyMobileMoneyProviders = await fetchLegacyMobileMoneyProviders()
+      const legacyMobileMoneyProviders = filterLegacyMobileMoneyProviders(
+        rawLegacyMobileMoneyProviders,
+        flow,
+        userLocation.countryCodeAlpha2,
+        digitalAsset
+      )
+
+      return { providers, legacyMobileMoneyProviders }
+    } catch (error) {
+      dispatch(showError(ErrorMessages.PROVIDER_FETCH_FAILED))
+    }
+  }, [])
+
+  const quoteOnPress = ({ quote, provider }: CicoQuote) => () => {
+    ValoraAnalytics.track(FiatExchangeEvents.provider_chosen, {
+      flow,
+      provider: provider.name,
+    })
+
+    if (quote && userLocation?.ipAddress && isSimplexQuote(quote)) {
+      navigate(Screens.Simplex, {
+        simplexQuote: quote,
+        userIpAddress: userLocation.ipAddress,
+      })
+      return
+    }
+
+    ;(quote as ProviderQuote).url && navigateToURI((quote as ProviderQuote).url)
+  }
+
+  if (asyncProviders.loading) {
+    return (
+      <View style={styles.activityIndicatorContainer}>
+        <ActivityIndicator size="large" color={colors.greenBrand} />
+      </View>
+    )
+  }
+  const activeProviders = asyncProviders.result?.providers
+
+  const cicoQuotes: CicoQuote[] =
+    getQuotes(activeProviders)
+      ?.filter(({ quote }) => (flow === CICOFlow.CashIn ? quote.cashIn : quote.cashOut))
+      .sort(sortQuotes) || []
+  return (
+    <ScrollView>
+      <PaymentMethodSection
+        cicoQuotes={cicoQuotes}
+        paymentMethod={PaymentMethod.Card}
+        setNoPaymentMethods={setNoPaymentMethods}
+        quoteOnPress={quoteOnPress}
+      />
+      <PaymentMethodSection
+        cicoQuotes={cicoQuotes}
+        paymentMethod={PaymentMethod.Bank}
+        setNoPaymentMethods={setNoPaymentMethods}
+        quoteOnPress={quoteOnPress}
+      />
+      <LegacyMobileMoneySection
+        provider={asyncProviders.result?.legacyMobileMoneyProviders[0]}
+        digitalAsset={digitalAsset}
+      />
+      <ExchangesSection selectedCurrency={route.params.selectedCrypto} flow={flow} />
```

Comment:
Could it ever be the case that no exchanges are available?

Reply:
Good question! I asked @lisaqyli if thats possible and she said there always is an option for SOME currency (usually CELO). If no exchanges are available the exchange screen actually encourages you to switch currency and try again. I think this is probably fine for now, but definitely could be optimized